test/test_pecora.jl

```diff
@@ -1,65 +1,108 @@
-using DelayEmbeddings, PyPlot, DynamicalSystemsBase, DelimitedFiles
+using DelayEmbeddings, DynamicalSystemsBase, DelimitedFiles
+using DifferentialEquations
 using Distances
 using Random
-UNDERSAMPLING = false
+using Test
+using Peaks
+using BenchmarkTools
 
-# %% Timeseries case
+
+@testset "Pecora" begin
+# %% Generate data
 Random.seed!(414515)
-desktop!()
-data = readdlm("data-lorenz.txt")
-s = data[:, 2] # input timeseries = first entry of lorenz
+lor = Systems.lorenz(ρ=60)
+data = trajectory(lor, 1280; dt=0.02, Ttr = 10)
 metric = Chebyshev()
-figure()
-ax1 = subplot(211)
-ylabel("⟨ε★⟩")
-ax2 = subplot(212)
-xlabel("τ (index units)")
-ylabel("⟨Γ⟩")
 
-optimal_τ = estimate_delay(s, "mi_min")
+UNDERSAMPLING = false
 
+@testset "Pecora univariate" begin
+# %% Timeseries case
+
+s = data[:, 2] # input timeseries = first entry of lorenz
+optimal_τ = estimate_delay(s, "mi_min")
 Tmax = 100
+K = 14
+samplesize = 0.05
+
+
+# using PyPlot
+# pygui(true)
+# figure()
+# ax1 = subplot(211)
+# ylabel("⟨ε★⟩")
+# ax2 = subplot(212)
+# xlabel("τ (index units)")
+# ylabel("⟨Γ⟩")
+
 
 τs = (0,)
-@time es, Γs = pecora(s, τs; T = 1:Tmax, N = 100, metric = metric, undersampling = UNDERSAMPLING)
-ax1.plot(es, label = "τs = $(τs)")
-ax2.plot(Γs)
+@code_warntype pecora(s, τs; delays = 0:Tmax, w = optimal_τ, samplesize = samplesize, K = K, metric = metric, undersampling = UNDERSAMPLING)

```

Comment:
Here I get a similar issue as in #45 and #38. I can not figure out why it is not typestable :/